nbi/engine/native/launcher/windows/src/JavaUtils.c

```diff
@@ -125,58 +125,79 @@ DWORD isJavaCompatible(JavaProperties *currentJava, JavaCompatible ** compatible
 JavaVersion * getJavaVersionFromString(char * string, DWORD * result) {
     JavaVersion *vers = NULL;
     if(getLengthA(string)>=3) {
-        //hope that at least we "major.minor" : 1.5
-        if(string[1]=='.') {
-            char c = string[0];
+        char *p = string;
+        
+        // get major 
+        long major = 0;
+        while(p!=NULL) {
+            char c = p[0];
+            if(c>='0' && c<='9') {
+                major = (major) * 10 + c - '0';
+                p++;
+                continue;
+            }
+            else if(c=='.'){
+                p++;
+            }
+            else{
+                return vers;
+            }
+            break;
+        }
+        
+        // get minor 
+        long minor = 0;
+        while(p!=NULL) {
+            char c = p[0];
             if(c>='0' && c<='9') {
-                long major = c - '0';
-                c = string[2];
-                if(c>='0' && c<='9') {
-                    char *p = string + 3;
-                    long minor = c - '0';
-                    *result = ERROR_OK;
-                    vers = (JavaVersion*) LocalAlloc(LPTR, sizeof(JavaVersion));
-                    vers->major  = major;
-                    vers->minor  = minor;
-                    vers->micro  = 0;
-                    vers->update = 0;
-                    ZERO(vers->build, 128);
-                    if(p!=NULL) {
-                        if(p[0]=='.') { // micro...
+                minor = (minor) * 10 + c - '0';
+                p++;
+                continue;
+            }
+            break;
+        }
+        
+        *result = ERROR_OK;
+        vers = (JavaVersion*) LocalAlloc(LPTR, sizeof(JavaVersion));
+        vers->major  = major;
+        vers->minor  = minor;
+        vers->micro  = 0;
+        vers->update = 0;
+        ZERO(vers->build, 128);
+        
+        if(p!=NULL) {
```

Comment:
This condition can be removed. `p` is never `NULL` (bcs of length condition at the start). `*p` is either `'.'` for micro, or not: `0` (for the end of string), or other non-`.` value.